packages/core/src/helpers/auth.ts

```diff
@@ -76,3 +80,19 @@ export const getConfiguredAliases = (context: AuthContext) => {
   const [primaryAlias, ...secondaryAliases] = aliases;
   return { primaryAlias, secondaryAliases };
 };
+
+/**
+ * Get the state of current actor. This is useful for checking which screen
+ * to render: e.g. `getActorState(state).matches('confirmSignUp.edit').
+ */
+export const getActorState = (state: AuthMachineState) => {
```

Comment:
nit - I know that adding return types to functions isn't "necessary", but since the given return type isn't explicitly clear to someone looking at this without prior knowledge of what they're working with, what do you think about adding it here? Not sure how easy/difficult that would be with Xstate typings.

Reply:
That's fair. This one is very doable, and @ErikCH has provided similar feedback about typing this. I'll add it in!

The only annoyance is that each actor has different context so we have to union them: something like `SignInState | SignUpState | 'SignOutState' | ResetPWState`. Let me try this.

Reply:
Done here: 
https://github.com/aws-amplify/amplify-ui/blob/8030d30c0cc32bf72db49e6d46284e5412836abc/packages/core/src/types/authMachine.ts#L51-L59

and return type is specified now: https://github.com/aws-amplify/amplify-ui/blob/8030d30c0cc32bf72db49e6d46284e5412836abc/packages/core/src/helpers/auth.ts#L91-L101